vendor/src/github.com/docker/swarmkit/ca/server.go

```diff
@@ -233,8 +233,9 @@ func (s *Server) IssueNodeCertificate(ctx context.Context, request *api.IssueNod
 					},
 				},
 				Spec: api.NodeSpec{
-					Role:       role,
-					Membership: api.NodeMembershipAccepted,
+					Role:         role,
+					Membership:   api.NodeMembershipAccepted,
+					Availability: request.Availability,
```

Comment:
@aaronlehmann Currently the definition of `Availability` is embedded inside the `NodeSpec`:
```
message NodeSpec {
        Annotations annotations = 1 [(gogoproto.nullable) = false];

        enum Membership {
                option (gogoproto.goproto_enum_prefix) = false;

                PENDING = 0 [(gogoproto.enumvalue_customname) = "NodeMembershipPending"];
                ACCEPTED = 1 [(gogoproto.enumvalue_customname) = "NodeMembershipAccepted"];
        }

        enum Availability {
                option (gogoproto.goproto_enum_prefix) = false;

                // Active nodes.
                ACTIVE = 0 [(gogoproto.enumvalue_customname) = "NodeAvailabilityActive"];
                ...
        }
```

Do we want to move `Availability` outside of `NodeSpec` (change `NodeSpec`)? Or we could spin off a new message (maybe `NodeAvailability`?) to include `NodeSpec.Availability`, and keep current `NodeSpec` message intact, like the following?
```
message NodeAvailability {

        NodeSpec.Availability availability = 1;

}
```